src/main/java/org/carlspring/cloud/storage/s3fs/S3FileSystemProvider.java

```diff
@@ -502,23 +573,32 @@ public void createDirectory(Path dir,
         }
 
         // create bucket if necessary
-        Bucket bucket = s3Path.getFileStore().getBucket();
-        String bucketName = s3Path.getFileStore().name();
+        final Bucket bucket = s3Path.getFileStore().getBucket();
+        final String bucketName = s3Path.getFileStore().name();
+
         if (bucket == null)
         {
-            s3Path.getFileSystem().getClient().createBucket(bucketName);
+            //TODO: How to get s3fs_region property value?
+            final Region region = Region.US_EAST_1;
```

Comment:
This shouldn't need to be set. It should default to be obtained from the Region provider chain. You could fall back to US_EAST_1 if there is a failure to construct the client. https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/java-dg-region-selection.html